packages/server/modules/automate/services/tracking.ts

```diff
@@ -73,43 +87,77 @@ const onAutomationRunStatusUpdated =
     })
   }
 
-const onRunCreated = async ({
-  automation,
-  run: automationRun,
-  source
-}: {
-  automation: AutomationWithRevision
-  run: InsertableAutomationRun
-  source: RunTriggerSource
-}) => {
-  // all triggers, that are automatic result of an action are in a need to be tracked
-  switch (source) {
-    case RunTriggerSource.Automatic: {
-      const mp = mixpanel({ userEmail: undefined })
-      await mp.track('Automation Run Triggered', {
-        automationId: automation.id,
-        automationName: automation.name,
-        automationRunId: automationRun.id,
-        projectId: automation.projectId,
-        source
-      })
-      break
+const getUserEmailFromAutomationRun =
```

Comment:
@gjedlicska Tbh I hate that we're doing this haha - are we basically gonna have to handle all ways a run could be created (which involves all kinds of DB queries, even re-pulling the same run again just to get extra metadata) in this tracking service just to get a user email?

At that point, if it's such an absolute necessity for Benjamin to get this tracking info, I'd rather just have a `triggeredById` column which is a lot easier to fill on run creation, cause you already have all of the relevant metadata at hand.

Reply:
I am not fond of it either, but this value def doesn't belong to our object model. On a conceptual basis, all of us still agrees, that a run cannot be 1-1 matched to a user, for all the reasons we've discussed.

Yet, for tracking reasons, we'd want to do this.
I do not want to add this value to our database schema, so we have to pay this price.